sdcm/nemesis.py

```diff
@@ -3820,6 +3820,108 @@ def _shrink_cluster(self, rack=None):
         self.log.info("Cluster shrink finished. Current number of nodes %s", num_of_nodes)
         InfoEvent(message=f'Cluster shrink finished. Current number of nodes {num_of_nodes}').publish()
 
+    # TODO: add support for the 'LocalFileSystemKeyProviderFactory' and 'KmipKeyProviderFactory' key providers
+    # TODO: add encryption for a table with large partitions?
+
+    def disrupt_enable_disable_table_encryption_aws_kms_provider(self):
+        # TODO: add check for Scylla that it supports the feature
+        if self.cluster.params.get("cluster_backend") != "aws":
+            raise UnsupportedNemesis("This nemesis is supported only on the AWS cluster backend")
+        if not self.cluster.params.get("aws_instance_profile_name_db"):
```

Comment:
for now it's a specific one, `aws_instance_profile_name_db` default for backup won't work.

Reply:
I understand it, but is there a way to check it correctly?
For now we can check for the presence of the `kms` word in the role name...

I prefer to have it here to narrow down the list of configuration where it may get run without real support.

Reply:
It should be related to the instance profile at all, we should run it if it's an enterprise version that supports KMS.

We should fix what needs fixing to make sure credentials are in place, regardless which method is used.

We need to edit the default profile use to have KMS as well, or create a new profile that have both.

Reply:
Added checks for `enterprise` and it's version.
Left the check for profile as is considering the idea that we will have the creds part be in our default profile.